### PR TITLE
address test related to merge

### PR DESCRIPTION
**Pull request by ravishetye**

### CODE REVIEW

File: backend/src/test/java/ai/verta/modeldb/blobs/DiffAndMerge.java [hunk 1]
```diff
@@ -24,8 +24,13 @@ public void diffAndMerge(AutogenBlob a, AutogenBlob b) throws ModelDBException {
...
 
     // Reapplying the diff should not change the result
-    diffedB = DiffMerger.mergeBlob(diffedB, d);
-    assertEquals(newB, diffedB);
+    HashSet<String> conflictSet = new HashSet<String>();
+    diffedB = DiffMerger.mergeBlob(diffedB, d, conflictSet);
+    if (newA == null || newB == null) {
+      assertEquals(newB, diffedB);
+    } else {
+      assertFalse(conflictSet.isEmpty());
```
Comment (conradoverta) [inline backend/src/test/java/ai/verta/modeldb/blobs/DiffAndMerge.java]: Is there anything we can say about the content of this? Or do you think just testing that it has anything is enough? The goal should be to assert the strongest statement we can with as few assumptions as we can. But maybe building the assert would be too complex for this case?

File: backend/swagger_codegen.sh [hunk 2]
```diff
@@ -381,12 +388,17 @@ do
         ${type}Diff d = DiffComputer.compute$(echo $type | sed 's,Autogen,,')Diff(newA, newB);
 
         // Applying the diff on top of the original A should get original B
-        ${type}Blob diffedB = DiffMerger.merge$(echo $type | sed 's,Autogen,,')(newA, d);
+        ${type}Blob diffedB = DiffMerger.merge$(echo $type | sed 's,Autogen,,')(newA, d, new HashSet<String>());
         assertEquals(newB, diffedB);
 
+		HashSet<String> conflictSet = new HashSet<String>();
```
Comment (conradoverta) [inline backend/swagger_codegen.sh]: Nit: spaces

File: backend/src/test/java/ai/verta/modeldb/blobs/DiffAndMerge.java (hunk 1 above)
Comment (ravishetye) [reply backend/src/test/java/ai/verta/modeldb/blobs/DiffAndMerge.java]: the assert is difficult to generate due to the randomized nature of the inputs.

File: backend/src/main/java/ai/verta/modeldb/versioning/blob/diff/DiffComputer.java [hunk 3]
```diff
@@ -305,14 +305,14 @@ public static AutogenPythonEnvironmentDiff computePythonEnvironmentDiff(
                     a,
                     b,
                     AutogenPythonEnvironmentBlob::getConstraints,
-                    AutogenPythonRequirementEnvironmentBlob::getLibrary,
+                    e -> e.getLibrary() + e.getConstraint(),
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/versioning/blob/diff/DiffComputer.java:308]: @conradoverta : do the changes make sense to you? If Ihave different hashing in differ and merger then we have issues during merge around modified  being detected as added

Comment (conradoverta) [reply backend/src/main/java/ai/verta/modeldb/versioning/blob/diff/DiffComputer.java:308]: Yeah, it seems reasonable to me. I don't know what you mean by "different hashing" though.

**Review by conradoverta — APPROVED** [review]

File: backend/src/main/java/ai/verta/modeldb/versioning/blob/diff/DiffComputer.java (hunk 3 above)
Comment (ravishetye) [reply backend/src/main/java/ai/verta/modeldb/versioning/blob/diff/DiffComputer.java:308]: generates different key for the map

**Pull request merged**